presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionTreeUtils.java

```diff
@@ -49,7 +49,7 @@ private ExpressionTreeFilter() {}
         return Optional.empty();
     });
 
-    public static Function<Node, Optional<Expression>> aggregateExtractorFunction(FunctionRegistry functionRegistry)
+    public static Function<Node, Optional<FunctionCall>> getAggregateExtractorFunction(FunctionRegistry functionRegistry)
```

Comment:
Actually, marking methods final given that (1) they're already static and (2) the class itself is marked final doesn't make sense. The only advantage of marking a method final and static is to prevent a sub-class from hiding the method in the super class and that doesn't apply because of (2). So I won't mark methods final, just the class itself.

http://stackoverflow.com/questions/1743715/behaviour-of-final-static-method